ddtrace/contrib/celery/app.py

```diff
@@ -30,9 +30,6 @@ def patch_app(app, pin=None):
         # Patch method
         setattr(app, method_name, wrapt.FunctionWrapper(method, wrapper))
 
-    # patch the Task class if available
-    setattr(app, 'Task', patch_task(app.Task))
```

Comment:
This changes how we instrument Celery. It's an unavoidable change I needed to do this refactoring. After an investigation of Celery internals, this cached property was causing some problems.

Reply:
Should we detail this a bit in the PR? What are we fixing, and what would be the potential impact for celery integration users?

Reply:
We re-wrote the integration removing the instrumentation done via monkey-patching. The new approach uses Celery signals. This is the first PR that refactors our tests, while another PR will switch the implementation.